File: app/src/main/java/com/settlex/android/ui/dashboard/model/ServiceUiModel.java
package com.settlex.android.ui.dashboard.model;

import com.settlex.android.data.enums.TransactionServiceType;

/**
 * Data model representing a service item with display name and icon resource
 */
public class ServiceUiModel {
    private final String name;
    private final int iconResId;
    private final int cashbackPercentage;
    private final TransactionServiceType type;

    public ServiceUiModel(String name, int iconResId, int cashbackPercentage, TransactionServiceType type) {
        this.name = name;
        this.iconResId = iconResId;
        this.cashbackPercentage = cashbackPercentage;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public int getIconResId() {
        return iconResId;
    }

    public int getCashbackPercentage() {
        return cashbackPercentage;
    }

    public TransactionServiceType getType() {
        return type;
    }
}

File: app/src/main/java/com/settlex/android/presentation/dashboard/services/model/ServiceUiModel.kt
package com.settlex.android.presentation.dashboard.services.model

import com.settlex.android.data.enums.TransactionServiceType

/**
 * Data model representing a service item with display name and icon resource
 */
data class ServiceUiModel(
    val name: String,
    val iconResId: Int,
    val cashbackPercentage: Int = 0,
    val label: String?,
    val transactionServiceType: TransactionServiceType,
    val destination: ServiceDestination?
)